include/deal.II/fe/fe_values.h

```diff
@@ -2911,6 +2911,84 @@ class FEValuesBase : public Subscriptor
     bool quadrature_points_fastest = false) const;
   //@}
 
+  /// @name Cell degrees of freedom
+  //@{
+
+  /**
+   * Return an object that can be thought of as an array containing all
+   * indices from zero to `dofs_per_cell`. This allows to write code
+   * using range-based for loops of the following kind:
+   * @code
+   *   FEValues<dim> fe_values (...);
+   *
+   *   for (auto &cell : dof_handler.active_cell_iterators())
+   *     {
+   *       fe_values.reinit(cell);
+   *       for (const auto i : fe_values.local_dof_indices())
+   *         for (const auto j : fe_values.local_dof_indices())
+   *           ... do something for DoF indices (i,j) ...
+   *     }
+   * @endcode
+   * Here, we are looping over all degrees of freedom on all cells, with
+   * `i` and `j` taking on all valid indices for cell degrees of freedom, as
+   * defined by the finite element passed to `fe_values`.
```

Comment:
Fair enough. There's certainly no need to introduce any ambiguity here. Since it would a replacement for this
```
for (auto &cell : dof_handler.active_cell_iterators())
{
  fe_values.reinit(cell);

  for (unsigned int q=0; q<fe_values.n_quadrature_points; ++q)
    for (unsigned int i=0; i<fe_values.dofs_per_cell; ++i)
      for (unsigned int j=0; j<fe_values.dofs_per_cell; ++j)
        cell_matrix(i,j) += ...; // Do something for DoF indices (i,j) at quadrature point q
}
```
then maybe the name `local_cell_dof_indices()` is more appropriate and clear (wrt. intent of use)?
This could then be the updated example
```
for (auto &cell : dof_handler.active_cell_iterators())
{
  fe_values.reinit(cell);

  for (const auto q : fe_values.quadrature_point_indices())
    for (const auto i : fe_values.local_cell_dof_indices())
       for (const auto j : fe_values.local_cell_dof_indices())
        cell_matrix(i,j) += ...; // Do something for DoF indices (i,j) at quadrature point q
}
```

Reply:
What about just `dof_indices()`? It's a term we're not using elsewhere, so there's no potential for misunderstanding. It also matches the `quadrature_point_indices()` function you introduced earlier.

Reply:
I got the feeling that this may be misinterpreted as the global DoF indices. Also, when using an `fe_face_values`, you still loop over all of the cell indices and not just the face indices. So I felt that this could do with some clarification. If you don't think that either of these are significant points of consideration, then I will adopt `dof_indices()` as you suggest.

Reply:
I'm ambivalent. I don't like the term `local_dof_indices` because we use that phrase with a different meaning elsewhere. I think `dof_indices()` is nice because it matches your use of `quadrature_point_indices()`. I could also go with `cell_dof_indices()`, but I still like `dof_indices()` best, I think.

@dealii/dealii : what do you all think?

Reply:
It's all good, I'm changing it to `dof_indices`.